corehq/apps/userreports/expressions/specs.py

```diff
@@ -331,3 +331,45 @@ def _get_subcases(self, case_id, context):
         subcases = [c.to_json() for c in CaseAccessors(domain).get_reverse_indexed_cases([case_id])]
         context.set_cache_value(cache_key, subcases)
         return subcases
+
+
+class GpsValueExpressionSpec(JsonObject):
+    LATITUDE = 'latitude'
+    LONGITUDE = 'longitude'
+    ELEVATION = 'elevation'
+    ACCURACY = 'accuracy'
+    SUPPORTED_GPS_VALUES = [LATITUDE, LONGITUDE, ELEVATION, ACCURACY]
+
+    type = TypeProperty('gps_get_value')
+    gps_expression = DefaultProperty(required=True)
+    gps_type = StringProperty(required=True)
+
+    def configure(self, gps_expression):
+        self._gps_expression = gps_expression
+        if self.gps_type not in GpsValueExpressionSpec.SUPPORTED_GPS_VALUES:
+            raise BadSpecError(_("gps_type'{}' is not valid. Valid options are: {} ").format(
```

Comment:
I used the same formatting as this code:
https://github.com/dimagi/commcare-hq/blob/477de9759cd140bee99293d0cc996102a53e2941/corehq/apps/userreports/expressions/list_specs.py#L64